core/spec/models/spree/unit_cancel/order_cancellations_recalculator_spec.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Spree::UnitCancel::OrderCancellationsRecalculator do
+  let(:inventory_unit) { create(:inventory_unit, line_item: line_item) }
+  let(:line_item) { create(:line_item) }
+  let(:unit_cancel) { Spree::UnitCancel.create!(inventory_unit: inventory_unit, reason: Spree::UnitCancel::SHORT_SHIP) }
+  let(:cancellation) { unit_cancel.adjust! }
+
+  before do
+    cancellation.update_columns(amount: 10, finalized: false)
+    cancellation.reload
+    line_item.reload
+  end
+
+  subject { described_class.new(line_item.order).call }
+
+  it "recalculates the cancellation adjustment to the correct amount" do
+    expect { subject }.to change { cancellation.reload.amount }.from(10).to(-20)
```

Comment:
Weird, actually. There's only one item to cancel, and only one canceled, yet the amount changes to -20? Turns out Unit Cancel adjustments are not even meant to be recalculated; they're always "finalized": https://github.com/solidusio/solidus/blob/main/core/app/models/spree/unit_cancel.rb#L27. Also, recalculating them does the wrong thing, as unit cancel adjustments do not take themselves into account: https://github.com/solidusio/solidus/blob/main/core/app/models/spree/unit_cancel.rb#L47

Should we make the cancellations recalculator a no-op by default then?

Reply:
> Should we make the cancellations recalculator a no-op by default then?

So the code accounting for when the adjustment is not finalized is just to keep the current behavior, which seems to never be called in real stores, and it's only a contrived scenario in the specs?

I was wondering that also in relation to the optimizations used to avoid the extra queries, are those done off the spec only too?

If the answer is yes to both I think a preference+deprecation cycle in which we raise if the cancelation adjustment is not finalized is in good order ✂️ ✂️ ✂️